packages/babel-plugin-transform-async-generator-functions/src/index.ts

```diff
@@ -83,6 +83,11 @@ export default declare(api => {
 
       path.traverse(yieldStarVisitor, state);
 
+      path.setData(
+        "babel-plugin-transform-async-generator-functions#async_generator_function",
+        true,
+      );
```

Comment:
https://github.com/babel/babel/blob/08b0472069cd207f043dd40a4d157addfdd36011/packages/babel-plugin-transform-async-generator-functions/src/index.ts#L103-L113
This is because we converted the `async generator function` in `Program.enter`.